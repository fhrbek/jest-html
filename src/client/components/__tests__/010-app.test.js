/* eslint-env jest */
/* eslint-disable global-require, import/newline-after-import */
import React from 'react';
import renderer from 'react-test-renderer';
import App, { socketDisconnect } from '../010-app';

// https://github.com/facebook/react/issues/7386#issuecomment-238091398
jest.mock('react-dom');
jest.mock('../015-appContents', () =>
  require('./mockComponent')('AppContents')
);

describe('App', () => {
  beforeEach(() => {
    require('whatwg-fetch');
  });

  it('renders correctly', () => {
    const match = {
      path: '/',
      url: '/',
      params: {},
    };
    const tree = renderer.create(<App match={match} location={{}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    socketDisconnect();
  });
});
