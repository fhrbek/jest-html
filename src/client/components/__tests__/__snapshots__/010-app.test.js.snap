// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  dataMockType="AppContents"
  error={null}
  fRedirectToRoot={false}
  fetchedItem={null}
  fetchedItemPath={null}
  fetchedItemType={null}
  location={Object {}}
  onRedirectToRoot={[Function]}
  saveAsBaseline={[Function]}
>
  <div
    style={
      Object {
        "color": "blue",
        "fontWeight": "bold",
      }
    }
  >
    AppContents
  </div>
</div>
------------HTML PREVIEW---------------
<div dataMockType="AppContents">
  <div style="color:blue;font-weight:bold;">
    AppContents
  </div>
</div>
`;
